bcipy/helpers/stimuli.py

```diff
@@ -243,26 +262,37 @@ def calibration_inquiry_generator(
     """
 
     len_alp = len(alp)
+    targets = []
```

Comment:
I found the logic of this function difficult to follow due to the amount of index juggling and list slicing. I also found it confusing that `stim_length` is used as the target index when there should not be a value. I think a lot of this could be cleaned up by explicitly declaring a sentinel value for the case where an inquiry should not display the target. Also, the `random` module has a lot of functions which can be helpful. Here is a proposed implementation.

```python
def calibration_inquiry_generator(
        alp: List[str],
        timing: List[float] = [0.5, 1, 0.2],
        color: List[str] = ['green', 'red', 'white'],
        stim_number: int = 10,
        stim_length: int = 10,
        stim_order: StimuliOrder = StimuliOrder.RANDOM,
        target_positions: TargetPositions = TargetPositions.RANDOM,
        nontarget_inquiries: int = 10,
        is_txt: bool = True) -> InquirySchedule:
    target_indexes = []
    no_target = None # sentinel value for inquiries that don't display a target
    if target_positions == target_positions.DISTRIBUTED:
        target_indexes = distributed_target_positions(stim_number, stim_length, nontarget_inquiries)
        # this line can be removed if `distributed_target_positions` uses None.
        target_indexes = [no_target if val == 10 else val for val in target_indexes]
    else:
        num_nontarget_inquiry = int((nontarget_inquiries / 100) * stim_number)
        num_target_inquiry = stim_number - num_nontarget_inquiry
        target_indexes = [no_target] * num_nontarget_inquiry
        target_indexes.extend(random.choices(range(stim_length), k=num_target_inquiry))
        random.shuffle(target_indexes)

    samples, times, colors = [], [], []

    for i in range(stim_number):
        inquiry = random.sample(alp, k=stim_length)
        if stim_order == StimuliOrder.ALPHABETICAL:
            inquiry = alphabetize(inquiry)

        target_index = target_indexes[i]
        if target_index is no_target:
            target = random.choice(list(set(alp) - set(inquiry)))
        else:
            target = inquiry[target_index]
        sample = [target, get_fixation(is_txt=is_txt), *inquiry]

        samples.append(sample)
        times.append([timing[i] for i in range(len(timing) - 1)] +
                     [timing[-1]] * stim_length)
        colors.append([color[i] for i in range(len(color) - 1)] +
                      [color[-1]] * stim_length)

    return InquirySchedule(samples, times, colors)
```